engineapi/backup_monitor.go

```diff
@@ -77,26 +102,91 @@ func NewBackupMonitor(logger logrus.FieldLogger,
 }
 
 func (m *BackupMonitor) monitorBackups() {
+	fallbackToExponentialBackoff := false
+
+	// Periodically liner timer
 	wait.PollUntil(BackupMonitorSyncPeriod, func() (done bool, err error) {
-		if err := m.syncBackups(); err != nil {
-			m.logger.Errorf("Stop monitoring because of %v", err)
+		fallback, err := m.syncBackups()
+		if err != nil && !fallback {
+			m.logger.Errorf("Stop monitoring because of %v in linear timer", err)
 			m.Close()
-			return false, err
+			return false, nil
+		}
+		if fallback {
+			fallbackToExponentialBackoff = fallback
+			return true, nil
 		}
 		return false, nil
 	}, m.ctx.Done())
+
+	if !fallbackToExponentialBackoff {
+		return
+	}
+
+	// Exponential back off timer
+	m.logger.Info("Fallback to exponential back off timer")
+	var (
+		initBackoff   = BackupMonitorSyncPeriod
+		maxBackoff    = 10 * time.Minute
+		resetDuration = BackupMonitorMaxRetryPeriod
```

Comment:
NIT: have you considered whether we can use an instance of a ratelimiter for all monitors?
Possibly either using one of these implementations or implementing this interface our selfs, I think having a ratelimiter that we can reuse / replace during unit tests would be a nice addition?

You can pass an instance of the rate limiter monitor as part of the construction of the monitor, so all monitors can use the same one, this way we can also add additional flow control (i.e. when 100 backups all need to be synced at the same time, etc)

REF: 
https://github.com/kubernetes/client-go/blob/master/util/workqueue/default_rate_limiters.go#L27

Reply:
Sounds good.

The only thing we need to do is manually calculated the number of maximum retries count since we use the `BackupMonitorMaxRetryPeriod` to determine how long should the monitor give up retry.
With the rate limiter interface, we use the maximum retries count to determine how many times should the monitor give up retry, like what we did for [backup lock deletion max retry count](https://github.com/longhorn/longhorn-manager/blob/452ca6e313b7378c9bd4c3d35e1a66a6eadfbcde/controller/backup_controller.go#L43).